internal/extsvc/github/globallock.go

```diff
@@ -0,0 +1,185 @@
+package github
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/redigo"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/internal/httpcli"
+	"github.com/sourcegraph/sourcegraph/internal/redispool"
+)
+
+var metricWaitingRequestsGauge = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "src_github_concurrency_lock_waiting_requests",
```

Comment:
Should all these alerts be explicit about it's only for github.com? i.e.
```suggestion
	Name: "src_githubcom_concurrency_lock_waiting_requests",
```